tests/test_pyrflx.py

```diff
@@ -1324,3 +1324,81 @@ def test_tlv_generating_tlv_error(tlv: MessageValue) -> None:
     tlv.set("Tag", "Msg_Error")
     assert tlv.valid_message
     assert tlv.bytestring == b"\xc0"
+
+
+def test_aspect_checksum(echo_request_reply_message: MessageValue) -> None:
+    # pylint: disable=protected-access
+    def checksum_icmp(message: bytes, **kwargs) -> int:
+        def add_ones_complement(num1: int, num2: int) -> int:
+            mod = 1 << 16
+            result = num1 + num2
+            return result if result < mod else (result + 1) % mod
+
+        tag_first, checksum_first_minus_one = kwargs.get(
+            "Tag'First .. Checksum'First - 1", (None, None)
+        )
+        identifier_first, data_last = kwargs.get("Identifier'First .. Data'Last", (None, None))
+        checksum_length = kwargs.get("Checksum'Length", None)
+        checksum_bytes = b""
+        if tag_first is not None and checksum_first_minus_one is not None:
+            checksum_bytes += message[tag_first : (checksum_first_minus_one + 1) // 8]
+        if checksum_length:
+            checksum_bytes += b"\x00" * (checksum_length // 8)
+        if identifier_first is not None and data_last is not None:
+            checksum_bytes += message[(identifier_first // 8) : (data_last + 1) // 8]
```

Comment:
As i understood it, the checksum should be calculated even if not all information needed to calculate it correctly are available (originally through the `preset_fields` method). Right now, the checksum is calculated every time a field has been set, that goes into the checksum calculation and if the checksum field itself is accessible. That leads to a situation where the checksum function is called but not all arguments needed to properly calculate the checksum are available.
We could alternatively:
1. Calculate the checksum only if all fields that go into its calculation have been set.
   (Pro: would only be calculated once in most cases; Con: if a field is changed after the checksum has been calculated, some logic is needed to calculate it again with the updated value)
2. Leave it as it is now, but check inside the user-defined checksum function if all needed arguments are available
(Pro: The checksum is always correct; Con: It is not very efficient to calculate a checksum without all information available, which involves a call to the  `_update_checksum_fields` method to simplify the expressions beforehand.)

Personally, i would prefer 1

Reply:
A checksum should be only calculated if all necessary information is available. The user-defined checksum function should expect to get all specified arguments and otherwise fail. So (1.) should be the correct approach.

Reply:
Done in 86ba388